### PR TITLE
IRIS-4921 | Handle content images in ogImage generation for Discussions

### PR DESCRIPTION
**Pull request by slayful**
## Links

* https://wikia-inc.atlassian.net/browse/IRIS-4921

### CODE REVIEW

File: server/app/lib/open-graph.js [hunk 1]
```diff
@@ -13,40 +12,41 @@ import {getStaticAssetPath} from './utils';
...
  */
-
-/**
- * @param {*} response
- **/
-function getThreadOpenGraph(response) {
-	const openGraph = response.payload._embedded.openGraph[0];
+function fromOpenGraph(openGraph) {
 	return {
 		image: openGraph.imageUrl,
```
Comment (rogatty) [inline server/app/lib/open-graph.js]: There is no `imageUrl` in JSDoc. Please update it.

File: server/app/lib/open-graph.js [hunk 2]
```diff
@@ -79,13 +79,26 @@ function generateFandomLogoOpenGraph(request) {
  * @param {*} context
  */
 function fetchOpenGraphImage(response, request, context) {
-	if (response.payload._embedded.openGraph) {
-		return getThreadOpenGraph(response);
+	const thread = response.payload._embedded;
+
+	if (thread.contentImages && thread.contentImages.length > 0) {
+		return fromContentImage(thread.contentImages[0]);
```
Comment (rogatty) [inline server/app/lib/open-graph.js]: I think we want the largest image, for both posts and replies.

Comment (rogatty) [reply server/app/lib/open-graph.js]: Oh, right, there is only one image in the post 😛

File: server/app/lib/open-graph.js [hunk 3]
```diff
@@ -79,13 +79,26 @@ function generateFandomLogoOpenGraph(request) {
  * @param {*} context
  */
 function fetchOpenGraphImage(response, request, context) {
-	if (response.payload._embedded.openGraph) {
-		return getThreadOpenGraph(response);
+	const thread = response.payload._embedded;
```
Comment (rogatty) [inline server/app/lib/open-graph.js]: In Ember we use post and reply. Can we stick to it? It'll be less confusing when introducing FastBoot.

File: server/app/lib/open-graph.js [hunk 4]
```diff
@@ -13,40 +12,41 @@ import {getStaticAssetPath} from './utils';
...
+function getLargestImage(images) {
+	images.sort((a, b) => b.imageWidth * b.imageHeight - a.imageWidth * a.imageHeight);
+	return images[0];
+}
+
+function getPostOpenGraphImages(posts) {
+	return posts
+		.map(post => post._embedded.openGraph)
+		.filter(openGraphs => openGraphs && openGraphs.length > 0)
+		.map(openGraphs => openGraphs[0])
```
Comment (rogatty) [inline server/app/lib/open-graph.js]: Method is named getPostOpenGraphImage**s** and we return an array. I don't think this line fits. The same goes for getPostContentImage**s**.

Comment (slayful) [reply server/app/lib/open-graph.js]: I believe there's a misunderstanding here - both open graph and content images are a list per post/reply in the response. At the moment BE returns up to one element in that list.

I was considering re-using find the biggest image code for selecting the open graph and content image from the list returned by BE, but that's neither necessary (up to one element for unforseeable future) nor the logic we'd end up choosing if/when we allow more than one content image or open graph per post/reply.

So this code filters posts without or an empty a list of opengraphs and then maps a list of one element into an element. Similar thing goes for content images.

Comment (rogatty) [reply server/app/lib/open-graph.js]: Yup, I thought we return an array with one item but there are actually as many items as posts with images.

**Pull request merged**